Postal/MailData+Decode.swift

```diff
@@ -26,136 +26,153 @@ import Foundation
 import libetpan
 
 extension MailData {
-    public var decodedData: NSData {
+    public var decodedData: Data {
         switch encoding {
-        case .encoding7Bit, .encoding8Bit, .binary, .other: return rawData
+        case .encoding7Bit, .encoding8Bit, .binary, .other: return rawData as Data
         case .base64: return rawData.base64Decoded
         case .quotedPrintable: return rawData.quotedPrintableDecoded
         case .uuEncoding: return rawData.uudecoded
         }
     }
 }
 
-extension NSData {
-    var base64Decoded: NSData {
+extension Data {
+    var base64Decoded: Data {
         return decodeWithMechanism(MAILMIME_MECHANISM_BASE64, partial: false).decoded
     }
     
-    var quotedPrintableDecoded: NSData {
+    var quotedPrintableDecoded: Data {
         return decodeWithMechanism(MAILMIME_MECHANISM_QUOTED_PRINTABLE, partial: false).decoded
     }
     
-    func decodeWithMechanism(mechanism: Int, partial: Bool) -> (decoded: NSData, remaining: NSData?) {
+    func decodeWithMechanism(_ mechanism: Int, partial: Bool) -> (decoded: Data, remaining: Data?) {
         var curToken: size_t = 0
-        var decodedBytes: UnsafeMutablePointer<Int8> = nil
+        var decodedBytes: UnsafeMutablePointer<Int8>? = nil
         var decodedLength: Int = 0
         let decodeFunc = partial ? mailmime_part_parse_partial : mailmime_part_parse
-        decodeFunc(UnsafePointer<Int8>(bytes), length, &curToken, Int32(mechanism), &decodedBytes, &decodedLength)
-        let decodedData = NSData(bytesNoCopy: decodedBytes, length: decodedLength) { (pointer, length) in
-            free(pointer)
+        
+        // (_ message: UnsafePointer<Int8>!, _ length: Int, _ indx: UnsafeMutablePointer<Int>!, _ encoding: Int32, _ result: UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>!, _ result_len: UnsafeMutablePointer<Int>!)
+        
+        let _ = withUnsafeBytes { (bytes: UnsafePointer<Int8>) in
```

Comment:
`winUnsafeBytes` returns the value of what you want. In this case the return value of the `decodedFunc` we didn't use in the previous version. Everything is ok there.